src/core/geometry/qgscompoundcurve.cpp

```diff
@@ -388,6 +388,22 @@ bool QgsCompoundCurve::isEmpty() const
   return true;
 }
 
+bool QgsCompoundCurve::isValid( QString &error SIP_OUT, int flags ) const
+{
+  if ( mCurves.isEmpty() )
+    return true;
+
+  for ( QgsCurve *curve : mCurves )
+  {
+    if ( !curve->isValid( error, flags ) )
+    {
+      error = QString( "At least one curve is not valid." );
+      return false;
+    }
+  }
+  return QgsCurve::isValid( error, flags );
```

Comment:
Will this one check if there is only one single point?

Reply:
Yes, there is a test.

A tiny python example:

```
l = QgsLineString([QgsPoint(0, 0)])
l.isValid()
# (False, 'LineString has less than 2 points and is not empty.')
c = QgsCompoundCurve()
c.addCurve(l)
c.isValid()
# (False, 'Curve[1]: LineString has less than 2 points and is not empty.')
```